src/components/data-model-table/editable.tsx

```diff
@@ -477,223 +478,279 @@ export const GenericDataTable: React.FC<GenericDataTableProps> = (
-              const gqlListQuery = gqlAssociationProps.gqlListQuery;
-              const gqlUseQueryVariablesKey =
-                gqlAssociationProps.gqlUseQueryVariablesKey;
-              let gqlQueryVariables = undefined;
-              if (
-                gqlUseQueryVariablesKey &&
-                gqlQueryVariablesMap &&
-                !!gqlQueryVariablesMap[gqlUseQueryVariablesKey]
-              ) {
-                if (
-                  typeof gqlQueryVariablesMap[gqlUseQueryVariablesKey] ===
-                  'function'
-                ) {
-                  gqlQueryVariables =
-                    gqlQueryVariablesMap[gqlUseQueryVariablesKey](record);
-                } else {
-                  gqlQueryVariables =
-                    gqlQueryVariablesMap[gqlUseQueryVariablesKey];
-                }
-              }
-              return (
-                <div className="editable-cell">
-                  <Form.Item name={field.name}>
-                    <AssociationSelection
-                      selectable={SelectionType.MULTIPLE}
-                      parentIdFieldName={parentIdFieldName!}
-                      associatedIdFieldName={associatedIdFieldName!}
-                      gqlQuery={gqlListQuery}
-                      parentRecord={record}
-                      gqlQueryVariables={gqlQueryVariables}
-                      associatedRecordClass={field.dtoClass!}
-                      value={form.getFieldValue(field.name)}
-                      onChange={(newValues: any[]) => {
-                        form.setFieldsValue({
-                          [field.name]: newValues,
-                        });
-                        setHasChanges(true);
-                      }}
-                      customActions={
-                        fieldAnnotations &&
-                        fieldAnnotations[field.name] &&
-                        fieldAnnotations[field.name].customActions
-                          ? fieldAnnotations![field.name].customActions
-                          : field.customActions
-                      }
-                    />
-                  </Form.Item>
-                </div>
-              );
-            }
-            if (
-              field.type === FieldType.nestedObject &&
-              field.gqlAssociationProps
-            ) {
-              let gqlAssociationProps;
-              if (
-                fieldAnnotations &&
-                fieldAnnotations[field.name]?.gqlAssociationProps
-              ) {
-                gqlAssociationProps =
-                  fieldAnnotations[field.name]?.gqlAssociationProps;
-              } else if (field.gqlAssociationProps) {
-                gqlAssociationProps = field.gqlAssociationProps;
-              }
-              if (!gqlAssociationProps) {
-                return (
-                  <Alert message="Missing gqlAssociationProps" type="error" />
+      ...schema
+        .map((field: FieldSchema) => {
```

Comment:
instead of returning null in the .map and then doing `.filter((obj) => !!obj),` - what do you think about adjusting this .map to instead use a simple `for of` loop and then we can just `continue` when there is a hidden field:
```
const allColumns = [actionsColumn];
for (let fieldSchema of schema) {
   ...
   if (hidden) continue;
   allColumns.push(...);
}
return allColumns;
```